### DISCUSSION TITLE
About this wonderful library

### DISCUSSION THREAD

**Discussion by mhsama** [AliDehbansiahkarbon/EasyDBMigrator#1] [General]
Hi Ali,
May I ask you to give us background experiences that made you create this wonderful library?

**Comment by AliDehbansiahkarbon** [comment]
Certainly, there are several reasons for this.

- Managing versioned Delphi classes is simpler than dealing with lengthy scripts. 
- Documenting the database structure in XML document format and inline comments is possible. 
- It's also possible to handle the versioning, upgrades, and downgrades of the database schema. 
- Additionally, it's safer than using a script file since it's integrated internally into the exe. 
- When working with multiple databases, migrations are a useful solution for evolving the database schema. These schema changes are saved in Delphi classes and can be committed to a version control system. 
- The tool covers multiple databases such as Microsoft SQL SERVER, MySQL, MariaDB, PostgreSQL, and Oracle.
- And supports various Delphi versions including Delphi XE5, Delphi XE6, Delphi XE7, Delphi XE8, Delphi 10 Seattle, Delphi 10.1 Berlin, Delphi 10.2 Tokyo, Delphi 10.3 Rio, Delphi 10.4 Sydney, and Delphi 11 Alexandria.
